src/main/java/com/pingcap/tikv/region/RegionManager.java

```diff
@@ -216,12 +222,44 @@ public void updateLeader(long regionId, long storeId) {
 
   /**
    * Clears all cache when a TiKV server does not respond
+   *
    * @param regionId region's id
-   * @param storeId TiKV store's id
+   * @param storeId  TiKV store's id
    */
   public void onRequestFail(long regionId, long storeId) {
     cache.invalidateRegion(regionId);
     cache.invalidateAllRegionForStore(storeId);
+    incrementCacheAccumulator(regionId, storeId, CacheInvalidateEvent.CacheType.REGION_STORE);
+  }
+
+  /**
+   * Used for notifying Spark driver to invalidate cache from Spark workers.
+   */
+  public void incrementCacheAccumulator(long regionId, long storeId, CacheInvalidateEvent.CacheType type) {
+    if (accFunc != null) {
+      accFunc.apply(new CacheInvalidateEvent(
+          regionId, storeId,
+          true, true,
+          type));
+    }
+  }
+
+  public void incrementRegionAccumulator(long regionId) {
```

Comment:
I think regionManager doesn't need to involve in this. Confine it inside error handler might be better because it indeed is a error event callback procedure. If you do it here, you might design a replay procedure here as well which make a RegionManager  listen to remote peer and change. I prefer move this into ErrorHandler which is simple. What's you opinion?

Reply:
100 percent agree! Will change it right away.